substra/sdk/backends/remote/backend.py

```diff
@@ -109,24 +111,134 @@ def add(self, spec, exist_ok=False, spec_options=None):
                 exist_ok=exist_ok,
                 spec_options=spec_options,
             )
+        elif asset_type == schemas.Type.ComputePlan:
+            # Compute plan auto batching feature
+            return self.__auto_batching_compute_plan(
+                spec=spec,
+                exist_ok=exist_ok,
+                spec_options=spec_options,
+            )
 
         with spec.build_request_kwargs(**spec_options) as (data, files):
             response = self._add(asset_type, data, files=files, exist_ok=exist_ok)
         # The backend has inconsistent API responses when getting or adding an asset
         # (with much less data when responding to adds).
         # A second GET request hides the discrepancies.
-        if asset_type == schemas.Type.ComputePlan:
-            return response
-
         key = _get_asset_key(response)
         return self.get(asset_type, key)
 
+    def __fill_tuple_lists(
+        self,
+        spec,
+        tuple_graph,
+        traintuples,
+        aggregatetuples,
+        composite_traintuples,
+        testtuples_by_train_id,
+    ):
+        for elem_id, _ in tuple_graph:
+            if elem_id in traintuples:
+                spec.traintuples.append(traintuples[elem_id])
+            elif elem_id in aggregatetuples:
+                spec.aggregatetuples.append(aggregatetuples[elem_id])
+            elif elem_id in composite_traintuples:
+                spec.composite_traintuples.append(composite_traintuples[elem_id])
+            else:
+                raise Exception("ERROR")
```

Comment:
We should avoid raising errors using the `Exception`. It is a good practice to subclass the base class to create our own errors. Such exception classes already exist in the substra SDK and should be use. It allows the end user our the dev to catch it and to handle it gracefully if needed.

It seems that this exception should never occur; or if it is raised it will highlight a developper bug. If this is the case, using an assert would be appropriate.

Reply:
Indeed, this was for debugging purposes, I remove it.